api/src/State/ContentNode/StoryboardPersistProcessor.php

```diff
@@ -1,35 +1,31 @@
 <?php
 
-namespace App\DataPersister\ContentNode;
+namespace App\State\ContentNode;
 
-use App\DataPersister\Util\DataPersisterObservable;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\Post;
+use ApiPlatform\State\ProcessorInterface;
 use App\Entity\ContentNode\Storyboard;
 use App\InputFilter\CleanHTMLFilter;
 use App\InputFilter\CleanTextFilter;
 use Ramsey\Uuid\Uuid;
 
-class StoryboardDataPersister extends ContentNodeAbstractDataPersister {
-    /**
-     * @throws \ReflectionException
-     */
+class StoryboardPersistProcessor extends ContentNodePersistProcessor {
     public function __construct(
-        DataPersisterObservable $dataPersisterObservable,
+        ProcessorInterface $decorated,
         private CleanHTMLFilter $cleanHTMLFilter,
         private CleanTextFilter $cleanTextFilter
     ) {
-        parent::__construct(
-            Storyboard::class,
-            $dataPersisterObservable
-        );
+        parent::__construct($decorated);
     }
 
     /**
      * @param Storyboard $data
      */
-    public function beforeCreate($data): Storyboard {
-        $data = parent::beforeCreate($data);
+    public function onBefore($data, Operation $operation, array $uriVariables = [], array $context = []): Storyboard {
+        $data = parent::onBefore($data, $operation, $uriVariables, $context);
 
-        if (null === $data->data) {
+        if ($operation instanceof Post && null === $data->data) {
```

Comment:
When the business logic was the same for update & create or very similar, I tried to avoid creating separate processors but keep it one PersistProcessor. In this specific case, this made it  easier to reuse `sanitizeData`.

But yeah, we can also make 2 processors and then outsource `sanitizeData` into a trait or a separate class.

Reply:
## Core Meeting Decision

Namespacing: Provider and Processer are left inside state as is